packages/flutter/test/material/ink_well_test.dart

```diff
@@ -1332,4 +1332,78 @@ void main() {
       textDirection: TextDirection.ltr,
     ));
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/6751
+  testWidgets('When InkWell has a Key and changes position, splash should not stop', (WidgetTester tester) async {
+    const String name = 'foo';
+
+
+    Widget buildApp({bool onTapDownChangeWrap = false, bool onTapChangeWrap = false, bool useKey = true}) {
+      bool wrap = false;
+
+      return MaterialApp(home: Scaffold(
+        body: Center(
+          child: StatefulBuilder(
+            builder: (BuildContext context, void Function(void Function()) setState) {
+              Future<void> changeWrap() async{
+                await Future<void>.delayed(const Duration(milliseconds: 50));
+
+                setState(() { wrap = !wrap; });
+              }
+
+              Widget child = InkWell(
+                key: useKey ? const Key(name) : null,
+                onTapDown: (_) async {
+                  if (onTapDownChangeWrap)
+                    changeWrap();
+                },
+                onTap: () async {
+                  if (onTapChangeWrap)
+                    changeWrap();
+                },
+                child: Container(
+                  height: 320,
+                  alignment: Alignment.center,
+                  child: const Text(name),
+                ),
+              );
+
+              if (wrap) {
+                child = Container(
+                  child: child,
+                );
+              }
+
+              return child;
+            },
+          ),
+        ),
+      ));
+    }
+
+    Future<void> test({bool onTapDownChangeWrap = false, bool onTapChangeWrap = false, bool useKey = true}) async {
+      await tester.pumpWidget(buildApp(
+          onTapDownChangeWrap: onTapDownChangeWrap,
+          onTapChangeWrap: onTapChangeWrap,
+          useKey: useKey,
+      ));
+
+      if (onTapDownChangeWrap)
+        await tester.startGesture(tester.getRect(find.byType(InkWell)).center);
+      else
+        await tester.tap(find.byType(InkWell));
+      await tester.pump(const Duration(milliseconds: 60));
+
+      expect(
+        tester.renderObject<RenderProxyBox>(find.byType(PhysicalModel)).child,
+        useKey ? (paints..circle()) : isNot(paints..circle()),
+      );
+    }
+
+    await test(useKey: false, onTapChangeWrap: true);
```

Comment:
Prefer DAMP over DIY in unit tests when there's a conflict. In other words, write each "variation" as a separate `test`. In order to reuse widget code, you can define a testing widget private to this file. You'll probably want to briefly document its usage and parameters as well.

You might reuse test process code if really necessary, but statement such as `useKey ? (paints..circle()) : isNot(paints..circle()),` is unfriendly to test readers while not saving much at all.

Reply:
Makes sense, and I'll update it.